### PR TITLE
Add Win8+ versions of KTHREAD to NDK and enable NDK tests

### PR DESCRIPTION
**Pull request by tkreuzer**
* [XDK] Fix KWAIT_BLOCK definition for Win8+
* [NDK] Add Win8+ version of KTHREAD
* [NDK] Improve NDK tests for KTHREAD and add them to build

### CODE REVIEW

File: sdk/include/ndk/ketypes.h [hunk 1]
```diff
@@ -919,9 +919,178 @@ typedef struct _KTHREAD_COUNTERS
...
+    {
+        struct
+        {
+#if (NTDDI_VERSION >= NTDDI_WIN10) // since 6.4.9841.0
+            ULONG_PTR CrossThreadReleasable : 1;
+#else
+            ULONG_PTR Waiting : 1;
+#endif
+            ULONG_PTR Busy : 1;
+            ULONG_PTR Reserved : (sizeof(PVOID) - 3); // previously Spare
```
Comment (ThFabba) [inline sdk/include/ndk/ketypes.h]: I think you mean 8*sizeof(PVOID)?

File: sdk/include/ndk/ketypes.h [hunk 2]
```diff
@@ -1369,6 +1538,431 @@ typedef struct _KTHREAD
...
+#endif
+    PVOID FirstArgument;
+    PKTRAP_FRAME TrapFrame;
+
+    union
+    {
+        KAPC_STATE ApcState;
+        struct
+        {
+            UCHAR ApcStateFill[FIELD_OFFSET(KAPC_STATE, UserApcPending) + 1]; // 32bit: , 64bit: 43/0x2B
```
Comment (ThFabba) [inline sdk/include/ndk/ketypes.h]: Do you have RTL_SIZEOF_THROUGH_FIELD available here?
`RTL_SIZEOF_THROUGH_FIELD(KAPC_STATE, UserApcPending)` seems more logical

**Comment by ThFabba** [comment]
Oh, and you're using two different users (timokreuzer vs tkreuzer). Is that intentional?

**Comment by tkreuzer** [comment]
@ThFabba That was accidental, fixed my email and the other things you mentioned.

**Pull request merged**